com.chartboost.mediation/Runtime/AdFormats/Fullscreen/ChartboostMediationFullscreenAdAndroid.cs

```diff
@@ -12,43 +12,48 @@ namespace Chartboost.AdFormats.Fullscreen
     /// <summary>
     /// Android implementation of IChartboostMediationFullscreenAd
     /// </summary>
-    public sealed class ChartboostMediationFullscreenAdAndroid : IChartboostMediationFullscreenAd
+    public sealed class ChartboostMediationFullscreenAdAndroid : ChartboostMediationFullscreenAdBase
     {
-        private readonly int _hashCode;
-        private bool _isValid = true;
         private readonly AndroidJavaObject _chartboostMediationFullscreenAd;
-        
-        public ChartboostMediationFullscreenAdAndroid(AndroidJavaObject fullscreenAd, ChartboostMediationFullscreenAdLoadRequest request)
+
+        public ChartboostMediationFullscreenAdAndroid(AndroidJavaObject fullscreenAd, ChartboostMediationFullscreenAdLoadRequest request) : base(uniqueId: fullscreenAd.HashCode())
         {
-            _hashCode = fullscreenAd.HashCode();
             _chartboostMediationFullscreenAd = fullscreenAd;
             var bidInfoMap = _chartboostMediationFullscreenAd.Get<AndroidJavaObject>("winningBidInfo");
             if (bidInfoMap != null)
                 WinningBidInfo = bidInfoMap.MapToWinningBidInfo();
             Request = request;
             LoadId = _chartboostMediationFullscreenAd.Get<string>("loadId");
-            CacheManager.TrackFullscreenAd(_hashCode, this);
+            CacheManager.TrackFullscreenAd(uniqueId.ToInt32(), this);
```

Comment:
Do we know this will always be a 32-bit Int? Just curious. No action is required.

Reply:
Actually, it should probably be int64 since the __bridge void * conversion from iOS native returns a long value not an int. 🤔 Casting it as an int can cause issues.